src/js/gutenberg-plugin.js

```diff
@@ -54,6 +55,130 @@ const ClassifAIToggle = () => {
 	);
 };
 
+/**
+ * Callback function to handle API response.
+ *
+ * @param {Object} resp
+ */
+const buttonClickCallBack = async ( resp ) => {
+	const { select, dispatch } = wp.data;
+	const postId = select( 'core/editor' ).getCurrentPostId();
+	const postType = select( 'core/editor' ).getCurrentPostType();
+	const postTypeLabel =
+		select( 'core/editor' ).getPostTypeLabel() || __( 'Post', 'classifai' );
+
+	if ( resp && resp.terms ) {
+		let updateNeeded = false;
+		const taxonomies = Object.keys( resp.terms );
+		const taxTerms = {};
+		taxonomies.forEach( ( taxonomy ) => {
+			let tax = taxonomy;
+			if ( 'post_tag' === taxonomy ) {
+				tax = 'tags';
+			}
+			if ( 'category' === taxonomy ) {
+				tax = 'categories';
+			}
+
+			const currentTerms =
+				select( 'core/editor' ).getEditedPostAttribute( taxonomy ) ||
+				[];
+			const newTerms = [
+				...currentTerms,
+				...resp.terms[ taxonomy ].map( ( term ) =>
+					Number.parseInt( term )
+				),
+			].filter( ( ele, i, a ) => a.indexOf( ele ) === i );
+
+			if ( newTerms && newTerms.length ) {
+				updateNeeded = true;
+				taxTerms[ tax ] = newTerms;
+			}
+		} );
+
+		if ( updateNeeded ) {
+			// Check for edited values in post.
+			const isDirty = await select( 'core/editor' ).isEditedPostDirty();
+			await dispatch( 'core' ).editEntityRecord(
+				'postType',
+				postType,
+				postId,
+				taxTerms
+			);
+			// If no edited values in post trigger save.
+			if ( ! isDirty ) {
+				await dispatch( 'core' ).saveEditedEntityRecord(
+					'postType',
+					postType,
+					postId
+				);
+			}
+
+			// Display success notice.
+			dispatch( 'core/notices' ).createSuccessNotice(
+				`${ postTypeLabel } ${ __(
+					'classified successfully.',
+					'classifai'
+				) }`,
```

Comment:
This isn't translatable as it presumes a language structure that may not apply to some languages.

```suggestion
				sprintf(
					/** translators: %s is post type label. */
					__( '%s classified successfully.', 'classifai' ),
					postTypeLabel
				) }`,
```

Please make sure `sprintf` is imported from `wp.i18n` and check my code, I haven't verified it for typos. 

To reduce the number of strings for translators, this duplicates the PHP version of the string:

https://github.com/10up/classifai/blob/919f3b915d39e12009c7d85888371055e6fb0a22/includes/Classifai/Admin/SavePostHandler.php#L232-L235

Reply:
Thanks a lot for noticing this.